tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -5,48 +5,139 @@ const sharingHelper = require('../helpers/sharingHelper')
       .openPublicLinkDialog(resource)
-      .addNewLink(settings.role)
+    return client.page.FilesPageElement.publicLinksDialog().addNewLink(settings)
   }
 )
 
 When('the public uses the webUI to access the last public link created by user {string}', async function (linkCreator) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
   return client.page.publicLinkFilesPage().navigateAndWaitTillLoaded(lastShareToken)
 })
 
-When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+When('last public share of the user {string} is deleted using API', async function (sharer) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  const lastShareId = lastShare.id
+  return sharingHelper.deleteLastPublicShare(sharer, lastShareId)
+})
 
-  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
+Then('user {string} should not have any public link',
+  async function (sharer) {
+    const resp = await sharingHelper.getAllShares(sharer)
+    assert.deepStrictEqual(
+      resp, [], 'User has no empty shares. Response: ' + resp)
+  })
 
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
+
+When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
+  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
   return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
 })
 
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('the public link of the user {string} should not have been generated', async function (sharer) {
+  const resp = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  assert.strictEqual(
+    resp, undefined, 'Error with public link shares! ' + resp
+  )
+})
+
+Then('the public should not get access to the publicly shared file', function () {
+  return client
+    .page
+    .publicLinkPasswordPage()
+    .assertResourceAccessDenied()
+})
+
+When('the user tries to edit the public link named {string} of file/folder/resource {string} changing following',
```

Comment:
Combine the following two steps together.
```suggestion
When('the user (tries to )edit/edits the public link named {string} of file/folder/resource {string} changing following',
```

Reply:
:confused: 
What do you mean?